level1a/handlers/level1a.py

```diff
@@ -37,39 +45,42 @@ def get_or_raise(variable_name: str) -> str:
     return var
 
 
-def get_last_date(
-    dataset: ds.FileSystemDataset,
-    start: Timestamp,
-) -> Optional[Timestamp]:
+def parse_event_message(event: Event) -> Tuple[str, str]:
     try:
-        dates = dataset.to_table(
-            filter=ds.field('EXPDate') >= start, columns=['EXPDate']
-        )['EXPDate'].to_pandas()
-    except pa.ArrowInvalid:
-        return None
-    if len(dates) == 0:
-        return None
-    return dates.max()
+        message: Dict[str, Any] = json.loads(event["Records"][0]["body"])
+        bucket = message["Records"][0]["s3"]["bucket"]["name"]
+        key = message["Records"][0]["s3"]["object"]["key"]
+    except (KeyError, TypeError):
+        raise InvalidMessage
+    return bucket, key
+
+
+def covers(
+    indices: DatetimeIndex,
+    first: Timestamp,
+    last: Timestamp,
+) -> bool:
+    return not (
+        len(indices) == 0
+        or first < indices.min()
+        or last > indices.max()
```

Comment:
```suggestion
        or first <= indices.min()
        or last >= indices.max()
```
or?

Reply:
I think it actually is fine as it was, at least if we want to allow the situation where `first == indices.min()`.
But I added a test case for this and rewrote the "double" negation for clarity.

Reply:
I saw that. I obviously misread it, so probably a good thing to rewrite it. ;)